### ISSUE TITLE
New Flutter project templates shouldn't reference dependencies that aren't in the standard Flutter package

### ISSUE THREAD

**Issue by strangedreamsNY** [flutter/flutter-intellij#6296]
The latest Flutter Project template in Android Studio contains a dependency reference to flutter_lint which isn't included as a part of the standard Flutter Installation. This can cause issues when attempting to start a new project while not connected to the internet since Android Studio doesn't support pub get in 'offline' mode and running flutter pub get --offline in Chipmunk builds isn't picked up by the IDE. This leaves you unable to use code inspection, build runners, or other dart lang features needed to work on your project. It would be helpful in the future to not include references to libraries or packages that aren't a part of the standard Flutter environment.

Flutter Plugin 70.0.2

Android Studio Chipmunk | 2021.2.1 Patch 2
Build #AI-212.5712.43.2112.8815526, built on July 10, 2022
Runtime version: 11.0.12+0-b1504.28-7817840 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 11.6.8
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 8
Registry: external.system.auto.import.disabled=true
Non-Bundled Plugins: Jetbrains TeamCity Plugin (2021.2.99602), Dart (212.5744), io.flutter (70.0.2)

[flutter_doctor.txt](https://github.com/flutter/flutter-intellij/files/9536175/flutter_doctor.txt)

**Comment by pq** [comment]
/fyi @devoncarew.

My thinking is, in general, we've converged on including lint references in project creation which does sort of require a `pub get` call. Is that right?

**Comment by devoncarew** [comment]
Yup, projects from both `dart create` and `flutter create` will require a `pub get` after initial creation. If working offline you could still create the projects, but would need to manually run a `dart pub get --offline` in order to finish setting them up.

**Comment by pq** [comment]
Awesome. Thanks for confirming!

I get that this may not be ideal @strangedreamsNY but as it's WAI, I'll close this out.

**Comment by strangedreamsNY** [comment]
After almost 3 years of no activity, I think it's safe to say this is pretty much what I expected. I still think it's a mistake and it, along with other unaddressed issues gives me pause for concern about the nature of Flutter's developer relations. I think I'm just used to other languages and platforms that don't require you to download additional packages and libraries just to compile a barebones project and this caught me by surprise with Flutter.  But, if that's how you guys think it should work, well, c'est la vie!

**Comment by pq** [comment]
I hear you. Sorry for the disappointment. Please bump any other unaddressed issues and we'll do our best to take a look!
